src/HDF5.jl

```diff
@@ -2023,36 +2011,34 @@ function h5s_get_simple_extent_dims(space_id::Hid)
     h5s_get_simple_extent_dims(space_id, dims, maxdims)
     return tuple(reverse!(dims)...), tuple(reverse!(maxdims)...)
 end
+# Note: The following two functions implement direct ccalls because the binding generator
+# cannot (yet) do the string wrapping and memory freeing.
 function h5t_get_member_name(type_id::Hid, index::Integer)
-    pn = ccall((:H5Tget_member_name, libhdf5), Ptr{UInt8}, (Hid, Cuint), type_id, index)
+    pn = ccall((:H5Tget_member_name, libhdf5), Ptr{Cchar}, (Hid, Cuint), type_id, index)
```

Comment:
Can we keep this as Ptr{UInt8} to be consistent with all other  string arguments we call from hdf5 with type `char *`

Reply:
Sure.